api/Controllers/CategoryController.cs

```diff
@@ -7,70 +7,76 @@
+    {
+      var categoryList = await _categoryService.GetCategoryList();
+      if (categoryList == null)
+      {
+        return NotFound();
+      }
+      var categoryDtoList = _mapper.Map<IList<CategoryDto>>(categoryList);
+      return Ok(_categoryService.TransformCategoryDtoListForOutput(categoryDtoList));
+    }
 
-        [HttpGet("{categoryId}")]
-        public async Task<IActionResult> GetCategory(int categoryId)
-        {
-            var category = await _categoryService.GetCategory(categoryId);
-            if(category == null) {
-                return NotFound();
-            }
-            var categoryDto = _categoryService.TransformCategoryDtoForOutput(_mapper.Map<CategoryDto>(category));
-            return Ok(categoryDto);
-        }
-        
-        [HttpGet]
-        public async Task<IActionResult> GetCategoryList() 
-        {
-            var categoryList = await _categoryService.GetCategoryList();
-            if(categoryList == null) {
-                return NotFound();
-            }
-            var categoryDtoList = _mapper.Map<IList<CategoryDto>>(categoryList);
-            return Ok(_categoryService.TransformCategoryDtoListForOutput(categoryDtoList));
-        }
+    [HttpPost]
+    public async Task<IActionResult> PostCategory(CategoryDto category)
+    {
+      Category newCategory = await _categoryService.CreateCategory(category);
 
-        [HttpPost]
-        public async Task<IActionResult> PostCategory(CategoryDto category) 
-        {   
-            Category newCategory = await _categoryService.CreateCategory(category);
-            
-            var createdCategory = await _categoryService.PostCategory(newCategory);
-            if(createdCategory == null) {
-                return BadRequest();
-            }
-            return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(createdCategory));
-        }
+      var createdCategory = await _categoryService.PostCategory(newCategory);
+      if (createdCategory == null)
+      {
+        return BadRequest();
+      }
+      return Created(nameof(GetCategory), _mapper.Map<CategoryDto>(createdCategory));
+    }
 
-        [HttpDelete]
-        public async Task<ActionResult> DeleteCategory(int categoryId) {
-            var removedCategory = await _categoryService.DeleteCategory(categoryId);
-            return removedCategory ? Ok() : BadRequest();
-        }
+    [HttpDelete]
+    public async Task<ActionResult> DeleteCategory(int categoryId)
+    {
+      var removedCategory = await _categoryService.DeleteCategory(categoryId);
+      return removedCategory ? Ok() : BadRequest();
+    }
 
-        [HttpPut]
-        public async Task<ActionResult> PutCategory(int categoryId, CategoryDto updatedCategory) {
-            Category categoryToPut = await _categoryService.CreateCategory(updatedCategory);
+    [HttpPut]
+    public async Task<ActionResult> PutCategory(int categoryId, CategoryDto updatedCategory)
```

Comment:
Thinking if we really need categoryId to be sent because it's equal to updatedCategory.id